ckanext/data_qld/command_functions.py

```diff
@@ -154,3 +155,112 @@ def send_email_dataset_overdue_notification():
     log.info('Started command SendEmailDatasetOverdueNotification')
     resource_freshness_helpers.process_email_notification_for_dataset_overdue()
     log.info('Finished command SendEmailDatasetOverdueNotification')
+
+
+def update_missing_values():
+    '''
+    Update datasets to set default values for metadata
+    '''
+    context = {'session': model.Session}
+    site_user = get_action('get_site_user')({'ignore_auth': True}, {})
+    context['user'] = site_user['name']
+
+    def _get_packages():
+        return model.Session.query(model.Package).filter(model.Package.state == 'active').all()
+
+    def _update_package(package_dict):
+        # Set some defaults
+        try:
+            print('Updating package {0}'.format(package_dict['id']))
+            get_action('package_patch')(context, package_dict)
+            return True
+        except Exception as ex:
+            print('Package exception: %s' % ex)
+            return False
+
+    def _update_resource(res_dict):
+        # Set some defaults
+        try:
+            print('Updating resource {0}'.format(res_dict['id']))
+            get_action('resource_patch')(context, res_dict)
+            return True
+        except Exception as ex:
+            print('Resource exception: %s' % ex)
+            return False
+
+    def _check_for_package_null_values(package):
+        package_patch = {"id": package.get('id')}
+
+        # All custom metadata will be in the extras
+        extras = package.get('extras', {})
+
+        if not extras.get('de_identified_data'):
+            package_patch['de_identified_data'] = 'NO'
+
+        if not extras.get('data_last_updated'):
+            # Set empty value to trigger package_patch which will call the validator data_qld_data_last_updated to set value from resources
+            package_patch['data_last_updated'] = ''
+
+        return package_patch
+
+    def _check_for_resource_null_values(res, package_patch):
+        resource_patch = {"id": res.get('id')}
+
+        if not res.get('nature_of_change'):
+            resource_patch['nature_of_change'] = 'edit-resource-with-no-new-data'
+
+        # Cannot use `not res.get('resource_visible')` because on subsequent runs the value could be 'FALSE' which will make the logic TRUE
```

Comment:
A string containing the value "FALSE" shouldn't be falsy; see https://www.pythontutorial.net/python-basics/python-boolean/

Reply:
You are correct, I have updated the logic.